cachito/workers/tasks/gomod.py

```diff
@@ -62,6 +72,34 @@ def _fail_if_bundle_dir_has_workspaces(bundle_dir: RequestBundleDir, subpaths: l
             raise InvalidRepoStructure("Go workspaces are not supported by Cachito.")
 
 
+def _fail_if_parent_replacement_not_included(packages_json_data: PackagesData) -> None:
+    """
+    Fail if any dependency replacement refers to a parent dir that isn't included in this request.
+
+    :param PackagesData packages_json_data: the collection of resolved packages for the request
+    :raises InvalidRequestData: if the module being replaced is not part of this request
+    """
+    modules = [
+        package["name"] for package in packages_json_data.packages if package["type"] == "gomod"
+    ]
+    log.debug(packages_json_data.packages)
+
+    for package in packages_json_data.packages:
+        for dependency in package.get("dependencies", []):
+            if dependency["version"] and ".." in Path(dependency["version"]).parts:
+                dep_normpath = os.path.normpath(
+                    os.path.join(package["name"], dependency["version"])
+                )
```

Comment:
I'm not 100% sure, but isn't the `dependency["version"]` relative to the **module** rather than the package?

To find the dep_normpath, I think you would first have to find the parent module of the *package*, the join the *module* name with the dependency version

Reply:
Yeah, I think you're right. When I was testing, the module and package were often the same, so I think I didn't notice